web_external/js/views/body/UploadDatasetView.js

```diff
@@ -257,10 +255,61 @@ isic.views.UploadDatasetView = isic.View.extend({
     }
 });
 
+isic.views.UploadDatasetRequestView = isic.View.extend({
+    events: {
+        'submit #isic-dataset-form': function (event) {
+            event.preventDefault();
+            this.$('#isic-dataset-submit').prop('disabled', true);
+            this.requestCreateDatasetPermission();
+        }
+    },
+
+    initialize: function (settings) {
+        this.render();
+    },
+
+    render: function () {
+        this.$el.html(isic.templates.uploadDatasetRequest({
+            user: girder.currentUser
+        }));
+
+        return this;
+    },
+
+    requestCreateDatasetPermission: function () {
+        girder.restRequest({
+            path: 'user/requestCreateDatasetPermission',
+            type: 'POST'
+        }).done(_.bind(function (resp) {
+            if (_.has(resp, 'extra') && resp.extra === 'hasPermission') {
+                // Directly update user permissions
+                girder.currentUser.setCanCreateDataset(true);
```

Comment:
When is this going to happen? If the user was granted permission to upload between the time that they last logged in or refreshed the front-end app and the time that they're clicking this button?